### PR TITLE
Add new v4 section and table of contents

### LINKED ISSUE #121: Add proper docs for full usage
We need proper documentation for:
 - how to use the new CLI
 - how to use the programmatic APIs
 - migration guide from v3 to v4

### PR DESCRIPTION
**Pull request by MilapNaik**
Closes #121

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,6 +48,26 @@ sample rokudeploy.json
...
+- `publish()` -> `sideload()`
+- `signExistingPackage()` -> `createSignedPackage()`
+- `deleteInstalledChannel()` -> `deleteDevChannel()`
+- `takeScreenshot()` -> `captureScreenshot()`
+
+Some functions were added which allow for any remote-to-Roku interaction: `keyPress()`, `keyUp()`, `keyDown()`, and `sendText()`
+
+Previously, functions `deploy()`, `createPackage()`, and `deployAndSignPackage()` were available in the Node API, but have been moved to CLI commands.
+
+Lastly, the default files array has changed. node modules and static analysis files have been excluded to speed up load times. Also, `fonts/` and `locale/` was added as they are in a few Roku documentation. The new default array can be seen in the section titled [Files Array](#files-array)
```
Comment (MilapNaik) [inline README.md:69]:
```suggestion
Lastly, the default files array has changed. node modules and static analysis files have been excluded to speed up load times. Also, `fonts/` and `locale/` was added as they are in some Roku documentation. The new default array can be seen in the section titled [Files Array](#files-array)
```

File: README.md [hunk 2]
```diff
@@ -172,23 +243,23 @@ rokuDeploy.createSignedPackage({
 ```
 
 Can't find what you need? Here are all of the public functions:
```
Comment (TwitchBronBron) [inline README.md:245]: Let's add a link to the src/rokudeploy.ts file

**Review by TwitchBronBron — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -243,23 +243,23 @@ rokuDeploy.createSignedPackage({
...
+- [closeChannel()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L226)
+- [sideload()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L239)
+- [rekeyDevice()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L379)
+- [createSignedPackage()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L426)
+- [deleteDevChannel()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L614)
+- [captureScreenshot()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L629)
+- [getOptions()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L698)
+- [checkRequiredOptions()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L738)
+- [getDeviceInfo()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L787)
+- [getDevId()](https://github.com/rokucommunity/roku-deploy/blob/v4/src/RokuDeploy.ts#L854)
```
Comment (TwitchBronBron) [inline README.md]: So, I was more thinking you would add a single link to `RokuDeploy.ts`, becuase these line numbers will change as we update the code. I think you can just list the functions without links, with a single link at the top saying "look through this file to see all the public methods available"

Comment (MilapNaik) [reply README.md]: Ohhh ok. Yeah the line numbers would change so one link sounds easier to maintain

Comment (MilapNaik) [reply README.md]: Done and also changed think to typescript interface for Roku Deploy Options

**Review by TwitchBronBron — APPROVED** [review]

**Pull request merged**